azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -0,0 +1,336 @@
+
+  // We want to limit the log sizes to about 20 megs
+  private final String jobLogChunkSize;
+  private final int jobLogNumFiles;
+  // If true, jobs will validate proxy user against a list of valid proxy users.
+  private final boolean validateProxyUser;
+
+
+  private FlowContainer(final Path projectDirPath, final Props props)
+          throws IOException {
+    this.projectDir = projectDirPath.toFile();
+
+    // Create Azkaban Props Map
+    this.azKabanProps = props;
+    // Setup global props if applicable
+    final String globalPropsPath = this.azKabanProps.getString("executor.global.properties", null);
+    if (globalPropsPath != null) {
+      this.globalProps = new Props(null, globalPropsPath);
+    }
+
+    // Setup DAO, a lot of it is redundant
+    final DataSource dataSource= new MySQLDataSource(this.azKabanProps,
+        new DBMetrics(new MetricsManager(new MetricRegistry())));
+
+    final DatabaseOperator dbOperator = new DatabaseOperator(
+        new QueryRunner(dataSource));
+
+    this.executorLoader = setupDao(dbOperator);
+
+    // project Loader
+    this.projectLoader = new JdbcProjectImpl(this.azKabanProps, dbOperator);
+
+    // setup executor service, TODO : revisit
+    this.executorService = Executors.newSingleThreadExecutor();
+
+    this.jobLogChunkSize = this.azKabanProps.getString("job.log.chunk.size", "5MB");
+    this.jobLogNumFiles = this.azKabanProps.getInt("job.log.backup.index", 4);
+    this.validateProxyUser = this.azKabanProps.getBoolean("proxy.user.lock.down", false);
+    this.jobTypeManager =
+        new JobTypeManager(
+            this.azKabanProps.getString(AzkabanExecutorServer.JOBTYPE_PLUGIN_DIR,
+                PluginManager.JOBTYPE_DEFAULTDIR),
+            this.globalProps, getClass().getClassLoader());
+    // Fetch keyStore props and use it get the KeyStore, put it in JobTypeManager
+    Props keyStoreLoadProps = this.jobTypeManager.getKeyStoreLoadProps();
+    if (keyStoreLoadProps != null) {
+      // Load HadoopSecurityManager
+      HadoopSecurityManager hadoopSecurityManager = null;
+      try {
+        final String hadoopSecurityClassName =
+                keyStoreLoadProps.getString(HadoopJobUtils.HADOOP_SECURITY_MANAGER_CLASS_PARAM);
+        final Class<?> hadoopSecurityManagerClass =
+                HadoopProxy.class.getClassLoader().loadClass(hadoopSecurityClassName);
+
+        logger.info("Loading hadoop security manager " + hadoopSecurityManagerClass.getName());
+        hadoopSecurityManager = (HadoopSecurityManager)
+                Utils.callConstructor(hadoopSecurityManagerClass, keyStoreLoadProps);
+      } catch (final Exception e) {
+        logger.error("Could not instantiate Hadoop Security Manager ", e);
+        throw new RuntimeException("Failed to get hadoop security manager!"
+                + e.getCause(), e);
+      }
+
+      final KeyStore keyStore = hadoopSecurityManager.getKeyStore(keyStoreLoadProps);
+      // Delete the cert file from disk
```

Comment:
Can you please add in comment that keystore is already loaded in above line. that is why file is not needed anymore.